cli/src/cluster_query.rs

```diff
@@ -1276,17 +1280,53 @@ pub fn process_show_stakes(
 
     let progress_bar = new_spinner_progress_bar();
     progress_bar.set_message("Fetching stake accounts...");
-    let all_stake_accounts = rpc_client.get_program_accounts(&solana_stake_program::id())?;
+
+    let all_stake_accounts = loop {
+        if let Some(vote_account_pubkeys) = vote_account_pubkeys {
+            // Use server-side filtering if only one vote account is provided
+            if vote_account_pubkeys.len() == 1 {
+                break rpc_client.get_program_accounts_with_config(
+                    &solana_stake_program::id(),
+                    RpcProgramAccountsConfig {
+                        filters: Some(vec![
+                            // Filter by `StakeState::Stake(_, _)`
+                            rpc_filter::RpcFilterType::Memcmp(rpc_filter::Memcmp {
+                                offset: 0,
+                                bytes: rpc_filter::MemcmpEncodedBytes::Binary(
+                                    bs58::encode([2, 0, 0, 0]).into_string(),
+                                ),
+                                encoding: Some(rpc_filter::MemcmpEncoding::Binary),
+                            }),
+                            // Filter by `Delegation::voter_pubkey`, which begins at byte offset 124
+                            rpc_filter::RpcFilterType::Memcmp(rpc_filter::Memcmp {
+                                offset: 124,
+                                bytes: rpc_filter::MemcmpEncodedBytes::Binary(
+                                    vote_account_pubkeys[0].to_string(),
+                                ),
+                                encoding: Some(rpc_filter::MemcmpEncoding::Binary),
+                            }),
```

Comment:
It was a little clunky to build this filter by hand...

Reply:
Could definitely add some tooling here, now that I'm not the only one using this!